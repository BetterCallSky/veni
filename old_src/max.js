import _makeError from './_makeError';

export default (max, msg) => (data) => {
  if (typeof data.value === 'string') {
    const stringMsg = msg || 'length must be less than or equal {{max}} characters long';
    if (data.value.length > max) {
      return {
        error: _makeError(data, stringMsg.replace('{{max}}', max)),
      };
    }
  }
  if (data.value > max) {
    const numberMsg = msg || 'must be less than or equal to {{max}}';
    return {
      error: _makeError(data, numberMsg.replace('{{max}}', max)),
    };
  }
  return undefined;
};
