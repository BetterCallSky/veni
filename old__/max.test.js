import max from '../src/max';

describe('max', () => {
  describe('string', () => {
    it('should return an error', () => {
      expect(max(2)({value: 'abc'})).toMatchSnapshot();
    });

    it('should return a custom message', () => {
      expect(max(2, 'too long')({value: 'abc'})).toMatchSnapshot();
    });

    it('should not return an error', () => {
      expect(max(4)({value: 'abc'})).toMatchSnapshot();
    });

    it('should not return an error (equal)', () => {
      expect(max(3)({value: 'abc'})).toMatchSnapshot();
    });
  });

  describe('number', () => {
    it('should return an error', () => {
      expect(max(2)({value: 5})).toMatchSnapshot();
    });

    it('should return a custom message', () => {
      expect(max(2, 'too big')({value: 5})).toMatchSnapshot();
    });

    it('should not return an error', () => {
      expect(max(4)({value: 3})).toMatchSnapshot();
    });

    it('should not return an error (equal)', () => {
      expect(max(3)({value: 3})).toMatchSnapshot();
    });
  });
});
