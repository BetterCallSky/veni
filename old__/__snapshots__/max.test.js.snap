// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`max number should not return an error (equal) 1`] = `undefined`;

exports[`max number should not return an error 1`] = `undefined`;

exports[`max number should return a custom message 1`] = `
Object {
  "error": Object {
    "error": "too big",
    "path": "value",
  },
}
`;

exports[`max number should return an error 1`] = `
Object {
  "error": Object {
    "error": "must be less than or equal to 2",
    "path": "value",
  },
}
`;

exports[`max string should not return an error (equal) 1`] = `undefined`;

exports[`max string should not return an error 1`] = `undefined`;

exports[`max string should return a custom message 1`] = `
Object {
  "error": Object {
    "error": "too long",
    "path": "value",
  },
}
`;

exports[`max string should return an error 1`] = `
Object {
  "error": Object {
    "error": "length must be less than or equal 2 characters long",
    "path": "value",
  },
}
`;
